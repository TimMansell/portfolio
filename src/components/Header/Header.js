import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import ShuffleCharacters from 'components/ShuffleCharacters';

import styles from './Header.module.scss';

export const Header = ({ primary, secondary, tertiary, text, title }) => {
  const classes = classnames(styles.heading, {
    [styles.headingPrimary]: primary,
    [styles.headingSecondary]: secondary,
    [styles.headingTertiary]: tertiary,
  });

  if (text) {
    text = <p data-test="heading-description">{text}</p>;
  }

  return (
    <div className={classes}>
      <ShuffleCharacters>
        <h2
          className={styles.title}
          data-test="heading-title"
          aria-label={title}
        >
          {title}
        </h2>
      </ShuffleCharacters>
      {text}
    </div>
  );
};

Header.propTypes = {
  primary: PropTypes.bool,
  secondary: PropTypes.bool,
  tertiary: PropTypes.bool,
  text: PropTypes.string,
  title: PropTypes.string.isRequired,
};

export default Header;
