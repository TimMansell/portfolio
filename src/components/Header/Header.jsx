import PropTypes from 'prop-types';
import classnames from 'classnames';

import styles from './Header.module.scss';

export const Header = ({ text, title, type }) => {
  const headerClasses = classnames(styles.heading, {
    [styles.headingPrimary]: type === 'primary',
    [styles.headingSecondary]: type === 'secondary',
    [styles.headingTertiary]: type === 'tertiary',
  });

  return (
    <div className={headerClasses}>
      <h2 className={styles.title} data-test="heading-title" aria-label={title}>
        {title}
      </h2>
      {text && (
        <p className={styles.description} data-test="heading-description">
          {text}
        </p>
      )}
    </div>
  );
};

Header.propTypes = {
  text: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['primary', 'secondary', 'tertiary']).isRequired,
};

export default Header;
