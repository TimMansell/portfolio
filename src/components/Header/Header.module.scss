@import 'variables';
@import 'utilities/pxtorem';

.heading {
  font-family: var(--font-family-secondary);
  margin-bottom: var(--vr-lg);
  text-align: center;

  @media (min-width: $bp-xxxl) {
    margin-bottom: var(--vr-xl);
  }

  &Primary {
    color: var(--primary-text-colour);
  }

  &Secondary {
    color: var(--secondary-text-colour);
    text-shadow: -1px -1px var(--colour-1-alt-2);
  }

  &Tertiary {
    color: var(--secondary-text-colour);
    text-shadow: -1px -1px var(--colour-2-alt-1);
  }
}

.title {
  font-size: pxtorem(48);
  line-height: 1.125;
  margin-bottom: var(--vr);
}
