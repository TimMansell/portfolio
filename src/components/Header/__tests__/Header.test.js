import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import Header from '../Header';

const props = {
  title: 'Test Title',
  text: 'text',
  type: 'primary',
};

describe('Header', () => {
  it('should render my component', () => {
    render(<Header {...props} />);
  });

  it('should match snapshot', () => {
    const { asFragment } = render(<Header {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('Should display a title', () => {
    const { getByText } = render(<Header {...props} />);

    expect(getByText(props.title)).toBeInTheDocument();
  });

  it('Should display a sub-title', () => {
    const { getByText } = render(<Header {...props} />);

    expect(getByText(props.text)).toBeInTheDocument();
  });

  it('Should display a primary class', () => {
    const { getByTestId } = render(<Header {...props} />);

    expect(getByTestId('header')).toHaveClass('headingPrimary');
  });

  it('Should display a secondary class', () => {
    const { getByTestId } = render(<Header {...props} type="secondary" />);

    expect(getByTestId('header')).toHaveClass('headingSecondary');
  });

  it('Should display a teritiary class', () => {
    const { getByTestId } = render(<Header {...props} type="tertiary" />);

    expect(getByTestId('header')).toHaveClass('headingTertiary');
  });
});
