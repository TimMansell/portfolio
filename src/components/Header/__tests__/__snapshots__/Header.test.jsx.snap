// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > should match snapshot 1`] = `
<div
  className="_heading_f4f7a7 _headingPrimary_f4f7a7"
>
  <h2
    aria-label="Test Title"
    className="_title_f4f7a7"
    data-test="heading-title"
  >
    Test Title
  </h2>
  <p
    className="_description_f4f7a7"
    data-test="heading-description"
  >
    text
  </p>
</div>
`;
